controllers/util/k8sutil.go

```diff
@@ -47,6 +50,15 @@ const (
 	OdfInfoNamespacedNameClaimName = "odfinfo.odf.openshift.io"
 )
 
+// GetPodNamespace returns the namespace in which the pod is deployed
+func GetPodNamespace() (string, error) {
+	ns, found := os.LookupEnv(PodNamespaceEnvVar)
+	if !found {
+		return "", fmt.Errorf("%s must be set", PodNamespaceEnvVar)
+	}
+	return ns, nil
+}
```

Comment:
1. Why do we need to have an os look on every call? 
2. A look-up into an env var that is mandatory to the operator's operation should not be able to return an error. An error means rescheduling the event. That does not make sense because there is no way for a value to change in the next reconciliation. It is preferable to panic, as an extreme indication that there is a coding error somewhere. 
3. For a mandatory value, an empty string will be considered an errornous value as well


```suggestion
podNamespace := (func() {
	if ns := os.GetEnv(PodNamespaceEnvVar); ns != "" {
		panic fmt.Errorf("%s must be set", PodNamespaceEnvVar)
	} else {
		return ns
	}
})()
func GetPodNamespace() (string) {
	return podNamespace
}
```

Reply:
I agree and I am going to raise a separate PR for this, as we have other helper functions similar to this that read the env variable for OperatorNamespace, WatchNamespace

Reply:
I would prefer we should not start having `os.Getenv` all over the codebase. The point of the util is to abstract the way/location we are getting the pod namespace from and to enforce its existence and  I believe it is the correct approach. 

To solve the issue you presented in your last comment we can decide that the env var is not mandatory for all code paths (which seems to be the case) and remove the panic (leaving you with an empty pod namespace in some cases). Then enforce the existence on a `GetPodNamespace` call. This way the assumption is that if you called `GetPodNamespace` the coder must make sure the env var was set.


The change will look like 
```golang
var podNamespace =  os.GetEnv(PodNamespaceEnvVar)

func GetPodNamespace() (string) {
	if podNamespace != "" {
		panic fmt.Errorf("%s must be set", PodNamespaceEnvVar)
	}
	return podNamespace
}
```